lib/ui/add_alert.dart

```diff
@@ -162,44 +170,100 @@ class _AddAlertPageState extends State<AddAlertPage> {
+                                  onPressed: () async {
+                                    Picker(
+                                        adapter: PickerDataAdapter<int>(
+                                            pickerData: [0,1,2,3,4,5,6,7,8,9,10,11,
+                                              12,13,14,15,16,17,18,19,20,21,22,
+                                              23,24,25,26,27,28,29,30,31]
+                                        ),
+                                        changeToFirst: true,
+                                        hideHeader: false,
+                                        onConfirm: (Picker picker, List value) {
+                                          setState(() {
+                                            daysToRepeat = picker.getSelectedValues()[0];
+                                          });
+                                        }
+                                    ).showModal(this.context);
+                                  },
+                                )
                               ],
-                        )),
+                            )),
+                        Offstage(
+                            offstage: !needToRepeat,
+                            child: Row(
+                              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                              crossAxisAlignment: CrossAxisAlignment.center,
+                              children: [
+                                TextButton(
+                                  child: Text('$hoursToRepeat Hour(s)'),
+                                  onPressed: () async {
+                                    Picker(
+                                        adapter: PickerDataAdapter<int>(
+                                            pickerData: [0,1,2,3,4,5,6,7,8,9,10,11,
+                                              12,13,14,15,16,17,18,19,20,21,22,
+                                              23,24]
+                                        ),
+                                        changeToFirst: true,
+                                        hideHeader: false,
+                                        onConfirm: (Picker picker, List value) {
+                                          setState(() {
+                                            hoursToRepeat = picker.getSelectedValues()[0];
+                                          });
+                                        }
+                                    ).showModal(this.context);
+                                  },
+                                ),
+                                TextButton(
+                                  child: Text('$minutesToRepeat Minute(s)'),
+                                  onPressed: () async {
+                                    Picker(
+                                        adapter: PickerDataAdapter<int>(
+                                            pickerData: [0,1,2,3,4,5,6,7,8,9,10,11,
```

Comment:
this is because user may needs to update some field back to zero, 

For example: user select 1 hour 1 min then they regret it and set back to 1 hour 0 min.

Reply:
I see, also we are already checking minute == 0 and display error